Throw exception from multicall methods

ruhan1 opened the pull request:
Fix the first part (throw exception) of NOS-1482.

dwalluck commented:
The issue is that the multicall methods return `null` instead of throwing `KojiClientException`, so there is no `KojiClientException` to catch there. The problem occurs further down the stack in `KojiClient. multiCall()`. 

I see the code is using `doXmlRpcAndWarn()` instead of `doXmlRpcAndThrow()`. It seems that one should not call `doXmlRpcAndWarn` without handling the fact that it can return `null`. But, I think it makes more sense to have it throw  `KojiClientException` like the other methods do.

I found two uses of `doXmlRpcAndWarn()`:

If you look at `KojiClient.java:1120`, it has `return response == null ? null : response;` (which is actually the same as `return response;`), which can be `null` in some cases.

It looks like `listArchivesForBuild()` has the same issue in that it returns `null` instead of throwing `KojiClientException`.

ruhan1 commented:
Updated. 
I replaced all the two doXmlRpcAndWarn with doXmlRpcAndThrow. This should be able to catch the XmlRpcTransportException from the doCall and throw to the upper layer.

dwalluck commented:
> Updated.
> I replaced all the two doXmlRpcAndWarn with doXmlRpcAndThrow. This should be able to catch the XmlRpcTransportException from the doCall and throw to the upper layer.

Thanks. So, why is the change to `enrichArchiveTypeInfo()` needed after this? Is there still a problem propagating the exception, or should the exception work without this change now?

ruhan1 commented:
The enrichArchiveTypeInfo changed because of the multicall signature changes (throwing new exception). The multicalls are in a lambda block and we have to use some way to throw exceptions too.
